Examples/Scripts/Python/propagation.py

```diff
@@ -18,43 +23,65 @@ def runPropagation(trackingGeometry, field, outputDir, s=None, decorators=[]):
 
     rnd = acts.examples.RandomNumbers(seed=42)
 
+    addParticleGun(
+        s,
+        ParticleConfig(num=1000, pdg=acts.PdgParticle.eMuon, randomizeCharge=True),
+        EtaConfig(-4.0, 4.0),
+        MomentumConfig(1 * u.GeV, 100 * u.GeV, transverse=True),
+        rnd=rnd,
+    )
+
+    # Run particle smearing
+    trackParametersGenerator = acts.examples.ParticleSmearing(
+        level=acts.logging.INFO,
+        inputParticles="particles_input",
+        outputTrackParameters="start_parameters",
+        randomNumbers=rnd,
+        sigmaD0=0.0,
+        sigmaZ0=0.0,
+        sigmaPhi=0.0,
+        sigmaTheta=0.0,
+        sigmaPtRel=0.0,
+    )
+    s.addAlgorithm(trackParametersGenerator)
+
     nav = acts.Navigator(trackingGeometry=trackingGeometry)
 
     stepper = acts.EigenStepper(field)
     # stepper = acts.AtlasStepper(field)
     # stepper = acts.StraightLineStepper()
-    # stepper = acts.SympyStepper(field)
 
-    print("We're running with:", type(stepper).__name__)
-    prop = acts.examples.ConcretePropagator(acts.Propagator(stepper, nav))
+    propagator = acts.examples.ConcretePropagator(acts.Propagator(stepper, nav))
 
-    alg = acts.examples.PropagationAlgorithm(
-        propagatorImpl=prop,
+    propagationAlgorithm = acts.examples.PropagationAlgorithm(
+        propagatorImpl=propagator,
         level=acts.logging.INFO,
-        randomNumberSvc=rnd,
-        ntests=1000,
-        sterileLogger=True,
-        propagationStepCollection="propagation-steps",
+        sterileLogger=False,
+        covarianceTransport=False,
+        inputTrackParameters="start_parameters",
+        outputPropagationSteps="propagation_steps",
     )
+    s.addAlgorithm(propagationAlgorithm)
 
-    s.addAlgorithm(alg)
+    # if not os.path.exists(outputDir + "/obj"):
+    #    os.makedirs(outputDir + "/obj")
 
-    # Output
-    s.addWriter(
-        acts.examples.ObjPropagationStepsWriter(
-            level=acts.logging.INFO,
-            collection="propagation-steps",
-            outputDir=outputDir + "/obj",
-        )
-    )
+    # s.addWriter(
+    #    acts.examples.ObjPropagationStepsWriter(
+    #        level=acts.logging.INFO,
+    #        collection="propagation_steps",
+    #        outputDir=outputDir + "/obj",
+    #    )
+    # )
```

Comment:
should we keep that?

Reply:
See above, I think for checking if the writer works - we should use the `test_writer.py` in my opinion (which we have), the root file is different as we check the hashes.